src/libopensc/sc.c

```diff
@@ -237,6 +237,12 @@ unsigned long lebytes2ulong(const u8 *buf)
 		| (unsigned long)buf[0];
 }
 
+void set_string(char **strp, const char *value)
+{
+	free(*strp);
+	*strp = value ? strdup(value) : NULL;
```

Comment:
Couls you surround this with `if (strp)`, just to be extra sure?

Reply:
Thanks. Added
```
if (strp == NULL) {
        return;
}
```
to avoid too much indentation.